src/ripple/rpc/handlers/AMMInfo.cpp

```diff
@@ -78,32 +78,54 @@ doAMMInfo(RPC::JsonContext& context)
 
     Issue issue1;
     Issue issue2;
+    uint256 ammID;
+    bool const isAssets =
+        params.isMember(jss::asset) && params.isMember(jss::asset2);
+    bool const isAMMAccount = params.isMember(jss::amm_account);
 
-    if (!params.isMember(jss::asset) || !params.isMember(jss::asset2))
+    if ((isAssets && isAMMAccount) || (!isAssets && !isAMMAccount))
     {
         RPC::inject_error(rpcINVALID_PARAMS, result);
         return result;
     }
 
-    if (auto const i = getIssue(params[jss::asset], context.j); !i)
-    {
-        RPC::inject_error(i.error(), result);
+    std::shared_ptr<ReadView const> ledger;
+    result = RPC::lookupLedger(ledger, context);
+    if (!ledger)
         return result;
+
+    if (isAssets)
+    {
+        if (auto const i = getIssue(params[jss::asset], context.j); !i)
+        {
+            RPC::inject_error(i.error(), result);
+            return result;
+        }
+        else
+            issue1 = *i;
```

Comment:
I see this pattern in this code a lot - where the error condition doesn't use the "if local" variable, and we handle the error condition first. I think this is simpler to read by handling the non-error case in the main branch and removing the `; !i` section. Here and other places as well.